elements/camera-capture/settings-pane.js

```diff
@@ -0,0 +1,291 @@
+    let panes = Array.from(bar.children);
+    panes.forEach(pane => {
+      pane.classList.add('hidden');
+    });
+  }
+
+  _toggleGroup(e) {
+    const id = e.target.getAttribute('for');
+    if (!id) return;
+
+    this.activePane = null;
+
+    if (id == "torch") {
+      const torch = this.shadowRoot.querySelector("#torch");
+      if (!torch.disabled) {
+        this.flash = !this.flash;
+        const constraints = { advanced: [{ torch: this.flash}] };
+        this.dispatchEvent(new CustomEvent('constraintschange', { detail: constraints }));
+      }
+      this.hide();
+      return;
+    }
+
+    const bar = this.shadowRoot.querySelector('#settings-bar');
+    const pane = this.shadowRoot.querySelector(`#${id}Settings`);
+
+    let panes = Array.from(bar.children).filter(el => el != pane);
+    panes.forEach(pane => {
+      pane.classList.add('hidden');
+    });
+
+    pane.classList.toggle('hidden');
+    if (!pane.classList.contains('hidden')) {
+      this.activePane = pane;
+      Array.from(pane.children).forEach(child => {
+        if ("layout" in child) child.layout();
+      })
+    }
+  }
+
+  _colorTemperatureChange(value) {
+    // See WB ranges: https://w3c.github.io/mediacapture-image/#white-balance-mode
+    let iconName = '';
+    switch (true) {
+      case (value >= 9000):
+        break;
+      case (value >= 8000): // Twilight mode.
+        iconName = 'brightness_3';
+        break;
+      case (value >= 6500): // Cloudy-daylight mode.
+        iconName = 'wb_cloudy';
+        break;
+      case (value >= 5500): // Daylight mode.
+        iconName = 'wb_sunny';
+        break;
+      case (value >= 5000):
+        break;
+      case (value >= 4000): // Fluorescent mode.
+        iconName = 'wb_iridescent';
+        break;
+      case (value >= 3500):
+        break;
+      case (value >= 2500): // Incandescent mode.
+        iconName = 'wb_incandescent';
+        break;
+      case (value == 0): // Continuous mode.
+        iconName = 'wb_auto';
+        break;
+    }
+    this.shadowRoot.getElementById('wbIcon').innerText = iconName;
+  }
+
+  firstUpdated() {
+    const controls = this.shadowRoot.querySelectorAll('settings-slider');
+    controls.forEach(control => {
+      control.onchange = e => {
+        const id = e.target.getAttribute('id');
+        const value = e.detail.value;
+        console.log(id, ":", value);
+
+        let constraints = { advanced: [{}] };
+        constraints.advanced[0][id] = value;
+        if (id == 'exposureTime') {
+          constraints.advanced[0]['exposureMode'] = 'manual';
+        } else if (id == 'focusDistance') {
+          constraints.advanced[0]['focusMode'] = 'manual';
+        } else if (id == 'colorTemperature') {
+          this._colorTemperatureChange(value);
+          constraints.advanced[0]['whiteBalanceMode'] = 'manual';
+        }
+
+        this.dispatchEvent(new CustomEvent('constraintschange', { detail: { constraints } }));
```

Comment:
Now the `constraintschange` change event is fired on mouse up, previously we changed it instantly on slider value change. Can we change the functionality to what it was ?

Reply:
done